automated/android/kunit/kunit.sh

```diff
@@ -0,0 +1,134 @@
+    # 20:43:20 stub: soc-utils-test.soc-utils#test_snd_soc_params_to_bclk: PASSED (0ms)
+    # 00:21:09 stub: kunit-example-test.example_init#example_init_test: PASSED (0ms)
+    #  | cut -d: -f4- \            # kunit-example-test.example_init#example_init_test: PASSED (0ms)
+    #  | tr -d ':' \               # kunit-example-test.example_init#example_init_test PASSED (0ms)
+    #  | awk '{print $1, $2}' \    # kunit-example-test.example_init#example_init_test PASSED
+    #  | sort | uniq \             # to filter out the duplication of FAILURE in Result Summary part
+    grep "stub:" "${f_kunit_log}" \
+        | cut -d: -f4- \
+        | tr -d ':' \
+        | awk '{print $1, $2}' \
+        | sort | uniq \
+        > "${f_kunit_stub_log}"
+    while read -r line; do
+        # kunit-example-test.example_init#example_init_test PASSED
+        # kunit-example-test.example#example_skip_test IGNORED
+        # soc-utils-test#soc-utils-test FAILURE
+        test_case_name=$(echo "${line}"|awk '{print $1}')
+        test_case_result=$(echo "${line}"|awk '{print $2}')
+
+        # reformat the test case name to avoid potential confusions
+        # being caused by some special characters
+        test_case_name=$(echo "${test_case_name}" \
+                            | tr -c '#@/+,[:alnum:]:.-' '_' \
+                            | tr -s '_' \
+                            | sed 's/_$//' \
+                            )
+
+        case "X${test_case_result}" in
+            "XPASSED")
+                report_pass "${test_case_name}"
+                ;;
+            "XIGNORED")
+                report_skip "${test_case_name}"
+                ;;
+            "XFAILURE")
+                report_fail "${test_case_name}"
+                ;;
+            *)
+                report_unknown "${test_case_name}"
+                ;;
+        esac
+    done < "${f_kunit_stub_log}"
+}
+
+while getopts "u:s:h" o; do
+  case "$o" in
+    u) TESTS_ZIP_URL="${OPTARG}" ;;
+    s) SQUAD_UPLOAD_URL="${OPTARG}" ;;
+    h|*) usage ;;
+  esac
+done
+
+# export ANDROID_SERIAL
+initialize_adb
+
+if [ -z "${TESTS_ZIP_URL}" ]; then
+    echo "The TESTS_ZIP_URL must be specified."
+    exit 1
+fi
+
+# download and unzip tests.zip
+rm -f "${F_TESTS_ZIP}" && \
+    curl -fsSL "${TESTS_ZIP_URL}" -o "${F_TESTS_ZIP}"
+rm -fr "${DIR_TESTS}" && \
+    mkdir -p "${DIR_TESTS}" && \
+    unzip -o "${F_TESTS_ZIP}" -d "${DIR_TESTS}"
+
+# clone the tradefed prebuilts repository
+rm -fr "${DIR_TF_PREBUILTS}" && \
+    git clone --depth 1 "${TRADEFED_PREBUILTS_GIT_URL}" "${DIR_TF_PREBUILTS}"
+
+# run the kunit test
+mkdir -p "${DIR_TEST_LOGS}"
+prebuilts/filegroups/tradefed/tradefed.sh  \
+    run commandAndExit \
+    template/local_min \
+    --template:map test=suite/test_mapping_suite \
+    --include-filter kunit \
+    --tests-dir="${DIR_TESTS}"  \
+    --log-file-path="${DIR_TEST_LOGS}" \
+    -s "${ANDROID_SERIAL}" |& tee "${F_KUNIT_LOG}"
+
+parse_kunit_log "${F_KUNIT_LOG}"
```

Comment:
you might want to do that in the postprocessing rather than in LAVA job. It dependes on how many results there are. If the number isn't too high, parsing in LAVA job is fine.

Reply:
Thanks for the check! Yes, that's one of my concern as well, but consider currently there are only about 500 test case, it should be OK, maybe it's still OK after 10 times increased.

Reply:
500 should be OK. I wouldn't go beyond 1k. LAVA is very slow when it comes to recording the results from the test job.